inst/include/dust/kernels.hpp

```diff
@@ -68,7 +71,13 @@ KERNEL void run_particles(size_t step_start,
 
     // Must only have a single __shared__ definition, cast to use different
     // types within it
-    real_t * shared_block_real = (real_t*)&shared_block[n_shared_int];
+    // Furthermore, writing must be aligned to the word length (may be an issue
+    // with int and real, as odd n_shared_int leaves pointer in the middle of an
+    // 8-byte word)
+    assert(sizeof(real_t) >= sizeof(int));
```

Comment:
```suggestion
    static_assert(sizeof(real_t) >= sizeof(int),
                  "Swap requires copying");
```

You might need to move this to the top level of the function though if it fails to compile. See https://en.cppreference.com/w/cpp/language/static_assert for details - we use this elsewhere in the package and you just get a compilation failure which is much nicer than a runtime failure